pyanaconda/modules/boss/installation.py

```diff
@@ -230,3 +290,489 @@ def run(self):
+        )
+
+        # add installation tasks for the Security DBus module
+        if is_module_available(SECURITY):
+            security_proxy = SECURITY.get_proxy()
+            security_dbus_tasks = security_proxy.InstallWithTasks()
+            os_config.append_dbus_tasks(SECURITY, security_dbus_tasks)
+
+        # add installation tasks for the Timezone DBus module
+        # run these tasks before tasks of the Services module
+        if is_module_available(TIMEZONE):
+            timezone_proxy = TIMEZONE.get_proxy()
+            timezone_dbus_tasks = timezone_proxy.InstallWithTasks()
+            os_config.append_dbus_tasks(TIMEZONE, timezone_dbus_tasks)
+
+        # add installation tasks for the Services DBus module
+        if is_module_available(SERVICES):
+            services_proxy = SERVICES.get_proxy()
+            services_dbus_tasks = services_proxy.InstallWithTasks()
+            os_config.append_dbus_tasks(SERVICES, services_dbus_tasks)
+
+        # add installation tasks for the Localization DBus module
+        if is_module_available(LOCALIZATION):
+            localization_proxy = LOCALIZATION.get_proxy()
+            localization_dbus_tasks = localization_proxy.InstallWithTasks()
+            os_config.append_dbus_tasks(LOCALIZATION, localization_dbus_tasks)
+
+        # add the Firewall configuration task
+        if conf.target.can_configure_network:
+            firewall_proxy = NETWORK.get_proxy(FIREWALL)
+            firewall_dbus_task = firewall_proxy.InstallWithTask()
+            os_config.append_dbus_tasks(NETWORK, [firewall_dbus_task])
+
+        configuration_queue.append(os_config)
+
+        # schedule network configuration (if required)
+        if conf.target.can_configure_network and conf.system.provides_network_config:
+            overwrite = payload.type in PAYLOAD_LIVE_TYPES
+            network_config = TaskQueue(
+                "Network configuration",
+                _("Writing network configuration"),
+                CATEGORY_SYSTEM
+            )
+            network_config.append(Task(
+                "Network configuration",
+                network.write_configuration,
+                (overwrite, )
+            ))
+            configuration_queue.append(network_config)
+
+        # add installation tasks for the Users DBus module
+        if is_module_available(USERS):
+            user_config = TaskQueue(
+                "User creation",
+                _("Creating users"),
+                CATEGORY_SYSTEM
+            )
+            users_proxy = USERS.get_proxy()
+            users_dbus_tasks = users_proxy.InstallWithTasks()
+            user_config.append_dbus_tasks(USERS, users_dbus_tasks)
+            configuration_queue.append(user_config)
+
+        # Anaconda addon configuration
+        addon_config = TaskQueue(
+            "Anaconda addon configuration",
+            _("Configuring addons"),
+            CATEGORY_SYSTEM
+        )
+
+        boss_proxy = BOSS.get_proxy()
```

Comment:
I don't think you should use API from the module this task is part of. That could cause deadlock. IIRC the module should never call itself API. Please pass the required values in as variable.

Reply:
According to https://docs.google.com/document/d/1_J81XYabaOJ2zyv92dZl6yz-h1M4uz2_b4cFxX2_0Ss/edit?tab=t.0#heading=h.njhxtfx5pgn3 calling for Task should work, I am not sure how strict we should be here.